File: arduinobot_cpp_examples/src/simple_publisher.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>
#include <chrono>

using namespace std::chrono_literals;

class SimplePublisher: public rclcpp::Node{
    public:
        SimplePublisher(): Node("simple_publisher"), counter_(0){
            publisher_ = create_publisher<std_msgs::msg::String>("simple_line_cpp", 10);
            timer_ = create_wall_timer(1s, std::bind(&SimplePublisher::timerCallback, this));

            RCLCPP_INFO(get_logger(), "Simple Publisher Started");
        }

        void timerCallback();

    private:
        unsigned int counter_;
        rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher_;
        rclcpp::TimerBase::SharedPtr timer_;
};

void SimplePublisher::timerCallback(){
    auto message = std_msgs::msg::String();
    message.data = "Simple Publisher publishing message " + std::to_string(counter_);
    publisher_->publish(message);
    counter_ += 1;
}

int main(int argc, char **argv){
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SimplePublisher>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: arduinobot_py_examples/arduinobot_py_examples/simple_publisher.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class SimplePublisher(Node):
    def __init__(self, name):
        super().__init__(name)

        # topic
        self.publish_topic = "simple_line"

        # publisher
        self.publisher_ = self.create_publisher(String, self.publish_topic, 10)

        # timer
        self.timer_period = 2.0

        # timer callback
        self.timer = self.create_timer(self.timer_period, self.timer_callback)

        # counter
        self.counter = 0

        self.get_logger().info("Simple Publisher Started")

    # Callback function
    def timer_callback(self):
        msg = String()
        msg.data = "Simple Publisher publishing message " + str(self.counter)
        self.publisher_.publish(msg)
        self.counter = self.counter + 1

    

def main():
    rclpy.init()                # Initialise ros2 communication
    simple_publisher = SimplePublisher("simple_publisher_node")
    rclpy.spin(simple_publisher) # keep the node up and ruuning
    simple_publisher.destroy_node()
    rclpy.shutdown()

if __name__ == "__main__":
    main()